junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/DisplayNameGenerator.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api.extension;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.lang.reflect.Method;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.util.ClassUtils;
+
+/**
+ * {@code DisplayNameGenerator} defines the {@link Extension} API for
+ * generating display names programmatically.
+ *
+ * @since 5.4
+ */
+@API(status = EXPERIMENTAL, since = "5.4")
+public interface DisplayNameGenerator extends Extension {
```

Comment:
This should not be an `Extension`. 

Although the deliverables for the issue state that it should be an "extension", the idea was not that it would actually be a JUnit Jupiter "extension" in the sense that it would receive an `ExtensionContext` or be registered like other extensions.